plugin/src/main/kotlin/com/moengage/internal/MavenPublishManager.kt

```diff
@@ -47,15 +48,29 @@ internal class MavenPublishManager(private val project: Project) {
         releaseVariant = project.findProperty(RELEASE_VARIANT) as? String ?: DEFAULT_RELEASE_VARIANT
         releasePortal = ArtifactReleasePortal.getMavenCentralPortal(project.findProperty(RELEASE_HOST) as? String)
 
-        val serviceBuilder =
-            ServiceBuilder(releasePortal, project.getUserName(releasePortal), project.getUserPassword(releasePortal))
+        val maxRetryCountOnTimeOut =
+            (project.findProperty(NEXUS_REPOSITORY_MAX_RETRY_ON_TIMEOUT) as? String)?.toInt() ?: 0
+        val networkTimeoutDuration = (project.findProperty(NETWORK_CALL_TIMEOUT_DURATION) as? String)?.toLong() ?: defaultTimeout
```

Comment:
should we have a maximum timeout value? waiting for infinite time or keeping the connection open for a very long time might not make sense.

Reply:
Added max timeout duration as 5 min